src/connection/server_support.c

```diff
@@ -2709,7 +2710,7 @@ css_process_new_slave (SOCKET master_fd)
 
   assert (ha_Server_state == HA_SERVER_STATE_TO_BE_ACTIVE || ha_Server_state == HA_SERVER_STATE_ACTIVE);
 
-  cubreplication::master_node::new_slave (new_fd);
+  cubreplication::replication_node_manager::get_instance ()->get_master_node ()->new_slave (new_fd);
```

Comment:
What about providing static methods as:
cubreplication::replication_node_manager::new_slave ()
cubreplication::replication_node_manager::add_ctrl_chn ()
..
which internally check the state of replication node and get the instance (taking into account concurency).